source/Module/STIG/Functions.Checklist.ps1

```diff
@@ -4,128 +4,217 @@
 
     .EXAMPLE
-        New-StigCheckList -ReferenceConfiguration $referenceConfiguration -ManualCheckFile "C:\Stig\ManualChecks\2012R2-MS-1.7.psd1" -XccdfPath $xccdfPath -OutputPath $outputPath
-        New-StigCheckList -ReferenceConfiguration $referenceConfiguration -ManualCheckFile $manualCheckFilePath -XccdfPath $xccdfPath -OutputPath $outputPath
+        Generate a checklist for multiple STIGs for an endpoint using DSC results obtained from Test-DscConfiguration, dehydrated/rehydrated using CLIXML:
+
+        $audit = Test-DscConfiguration -ComputerName localhost -MofFile 'C:\localhost.mof'
+        $audit | Export-Clixml 'C:\TestDSC.xml'
+        
+        $auditRehydrated = import-clixml C:\TestDSC.xml
+        $XccdfPath = 'C:\STIGS\SQL Server\U_MS_SQL_Server_2016_Instance_STIG_V1R7_Manual-xccdf.xml','C:\STIGS\Windows.Server.2012R2\U_MS_Windows_2012_and_2012_R2_DC_STIG_V2R19_Manual-xccdf.xml'
+        $outputPath = 'C:\SqlServer01_dsc.ckl'
+        $ManualChecklistEntriesFile = 'C:\ManualCheckListEntriesSqlServer01ExcelExport.xml'
+
+        New-StigCheckList -DscResults $auditRehydrated -XccdfPath $XccdfPath -OutputPath $outputPath -ManualChecklistEntries $ManualChecklistEntriesFile
 #>
 function New-StigCheckList
 {
     [CmdletBinding()]
-    [OutputType([xml])]
+    [OutputType([XML])]
     param
     (
         [Parameter(Mandatory = $true, ParameterSetName = 'mof')]
-        [string]
+        [Alias('MofFile')]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript(
+        {
+            If (Test-Path -Path $_ -PathType Leaf)
+            {
+                Return $True
+            } else {
+                Throw "$($_) is not a valid path to a reference configuration (.mof) file. Provide a full valid path and filename."
+            }
+        }
+        )]
+        [String]
         $ReferenceConfiguration,
 
-        [Parameter(Mandatory = $true, ParameterSetName = 'result')]
-        [psobject]
-        $DscResult,
+        [Parameter(Mandatory = $true, ParameterSetName = 'dsc')]
+        [PSObject]
+        $DscResults,
 
         [Parameter(Mandatory = $true)]
-        [string]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript(
+        {
+            foreach ($filename in $_)
+            {
+                If (Test-Path -Path $filename -PathType Leaf)
+                {
+                    Return $True
+                } else {
+                    Throw "$($filename) is not a valid path to a DISA STIG .xccdf file. Provide a full valid path and filename."
+                }
+            }
+        }
+        )]
+        [String[]]
         $XccdfPath,
 
-        [Parameter(Mandatory = $true)]
-        [System.IO.FileInfo]
-        $OutputPath,
-
         [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript(
+        {
+            If (Test-Path -Path $_ -PathType Leaf)
+            {
+                Return $True
```

Comment:
from a consistency perspective, $True should be $true.  There are other entries, this however was the first example.

Reply:
Done.